### ISSUE TITLE
install.pp failure on AWS

### ISSUE THREAD

**Issue by andrewwatson** [davidcollom/puppet-serf#8]
$osfamily evaluates as "Linux" on Amazon Linux boxes so I updated this case statement:

```
     13     case $::osfamily {
     14       'Debian': {
     15         $install_url = "https://dl.bintray.com/mitchellh/serf/${::serf::version}_linux_${::architecture}.zip"
     16       }
     17       'Linux': {
     18         $install_url = "https://dl.bintray.com/mitchellh/serf/${::serf::version}_linux_${::architecture}.zip"
     19       }
     20       default: {
     21         fail "Operating system ${::osfamily} ${::operatingsystem} is not supported yet."
     22       }
     23     }
```

but then when i try to use it like this:

```
      2 include serf
      3 
      4 class  { 'serf':
      5     bind            =>  $::ipaddress_eth1
      6     event_handler   =>  ['/root/bin/handler.sh']
      7 }
```

i get this error on the client

```
root@ip-172-31-8-10 ~]# puppet agent -t apply
err: Could not retrieve catalog from remote server: Error 400 on SERVER: no such file to load -- json
warning: Not using cache on failed catalog
err: Could not retrieve catalog; skipping run
```

any ideas?

**Comment by davidcollom** [comment]
Hi, what version of puppet are you running this against? i've seen that the JSON library isn't included/loaded within puppet 2.x

**Comment by andrewwatson** [comment]
2.7 i believe.

**Comment by davidcollom** [comment]
on my machines, performing a `gem install json` resolved this issue. give it a shot?

**Comment by andrewwatson** [comment]
My base amazon ami don't even have rubygems pkg though. I guess i could
install it via puppet.

**Comment by andrewwatson** [comment]
So I changed my cloud-init for my amazon image to run this script upon
instance creation:

#!/bin/bash

yum -y install puppet
yum -y install libselinux-ruby rubygem-json pciutils dmidecode

rpm -Uvh
https://yum.puppetlabs.com/el/6x/products/x86_64/puppet-3.4.2-1.el6.noarch.rpm\

https://yum.puppetlabs.com/el/6x/dependencies/x86_64/ruby-shadow-1.4.1-13.el6.x86_64.rpm\

https://yum.puppetlabs.com/el/6x/dependencies/x86_64/rlwrap-0.37-3.el6.x86_64.rpm\

https://yum.puppetlabs.com/el/6x/dependencies/x86_64/ruby-rgen-0.6.5-1.el6.noarch.rpm\

https://yum.puppetlabs.com/el/6x/products/x86_64/hiera-1.3.0-1.el6.noarch.rpm\

https://yum.puppetlabs.com/el/6x/products/x86_64/facter-1.6.18-1.el6.x86_64.rpm\

https://yum.puppetlabs.com/el/6x/dependencies/x86_64/ruby-augeas-0.4.1-1.el6.x86_64.rpm

puppet resource host puppet ensure=present ip=172.31.1.183
puppet resource service puppet ensure=running

and now i can get past the json issue but it goes to install
https://dl.bintray.com/mitchellh/serf/0.3.0_linux_x86_64.zip  and gets a
404!

Andy Watson
http://about.me/andrewwatson

Reach out to me with questions about your startup, technology and more
https://clarity.fm/#/andy

**Comment by davidcollom** [comment]
Hi Andrew,
Sorry for the delay in getting back to you, You can override the install_url 404 issue by specifying the exact/correct URL by:

```
class{'serf':
  install_url = 'https://dl.bintray.com/mitchellh/serf/0.3.0_linux_amd64.zip'
}
```

It appears that your distro / facter is returning x86_64 where by Ubuntu/Debian based systems are responding with amd_64 (as expected)

As the readme specifies, I've hardlies tested with anything other than Ubuntu/Debian based distro's I will attempt to fix these issues sometime this week and push an update for you.

Thank you for the feedback.

**Comment by davidcollom** [comment]
This was resolved at  54671987b846d404242a573f27a138ae6b7b9d34
